packages/block-editor/src/components/rich-text/index.native.js

```diff
@@ -589,7 +590,7 @@ function RichTextWrapper(
 			selectionEnd={ selectionEnd }
 			onSelectionChange={ onSelectionChange }
 			tagName={ tagName }
-			start={ start }
+			start={ Math.min( start, MAX_INT ) }
```

Comment:
Nice catch. I'll also look to see if we can handle the negative values better in the ordered list settings

Reply:
Looks like there is more going on with negative values in ordered lists which I feel is now out of scope for this issue. Instead I've opened this issue to capture those https://github.com/WordPress/gutenberg/issues/39827

Reply:
Gotcha. It indeed looks like the native mobile behavior is all over the place in regards to handling negative/zeros/large values.

I like how the current solution reflects the limit to the UI. That is, when you input a large value, the UI replaces it with the limit. That can still be confusing to the user and would be better with some more hinting to what happened there, but since it's the fairly infrequent cases of very large numbers, I don't think it's a big issue to block the PR on it.

What do you think about renaming the `minMaxInteger` function to something like `limitValue` and break it out into its own module to implement it in the `.native.js` variant only?